kivy/uix/behaviors/resize.py

```diff
@@ -0,0 +1,467 @@
+    the right resize border
+    It is switched to True when mouse is inside the right resize border and
+    False when it isn't.
+    It adjusts the mouse cursor and manages resizing when touch is moved.
+
+    :attr:`resizing_right` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to False.
+    '''
+
+    resizing_up = BooleanProperty(False)
+    '''A State which is enabled/disabled depending on the position relative
+    to the upper resize border
+    It is switched to True when mouse is inside the upper resize border and
+    False when it isn't.
+    It adjusts the mouse cursor and manages resizing when touch is moved.
+
+    :attr:`resizing_up` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to False.
+    '''
+
+    resizing_down = BooleanProperty(False)
+    '''A State which is enabled/disabled depending on the position relative
+    to the lower resize border
+    It is switched to True when mouse is inside the lower resize border and
+    False when it isn't.
+    It adjusts the mouse cursor and manages resizing when touch is moved.
+
+    :attr:`resizing_down` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to False.
+    '''
+
+    resizing = BooleanProperty(False)
+    '''State of widget resizing.
+    It is switched to True when a resize border is touched and back to
+    False when it is released.
+    It manages resizing when touch is moved.
+
+    :attr:`resizing` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to False.
+    '''
+
+    can_move_resize = BooleanProperty(True)
+    '''Move widget when resizing down or left.
+    To keep position on screen in a floatlayout,
+    actual postition has to be adjusted.
+    Resizing and changing position variables is problematic
+    inside movement restricting widgets,
+    (StackLayout, BoxLayout, others) this property manages that.
+
+    :attr:`can_move_resize` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to True.
+    '''
+
+    def __init__(self, **kwargs):
+        super(ResizableBehavior, self).__init__(**kwargs)
+        Window.bind(mouse_pos=lambda obj, val: self.on_mouse_move(val))
+        self.cursor = None
+        app = App.get_running_app()
+        try:
+            app.resizable_cursor
+        except AttributeError:
+            app.resizable_cursor = ModalViewModified()
+        self.modalview = app.resizable_cursor
+        self.cursor = self.modalview.cursor
+        self.oldpos = []
+        self.oldsize = []
+
+
+    def on_enter(self):
+        self.on_enter_resizable()
+
+    def on_leave(self):
+        self.cursor.hidden = True
+
+    def on_enter_resizable(self):
+        Window.show_cursor = False
+        self.cursor.hidden = False
+
+    def on_leave_resizable(self):
+        Window.show_cursor = True
+        self.cursor.hidden = True
+
+    def on_mouse_move(self, pos):
+        if self.hovering and self.cursor:
```

Comment:
I still think this:
```
if self.hovering and not self.resizing:
    if self.cursor:
        do_something()
    <the rest>
```
is prettier than the current alternative, the less "stairs" in code, the more readable it is.